src/service/handlers/prepared.rs

```diff
@@ -276,21 +276,78 @@ pub(crate) async fn do_put_prepared_statement_update(
         ));
     }
 
-    let parameter_sets = SwanFlightSqlService::collect_parameter_sets(request).await?;
+    // Collect RecordBatches first - we can only consume the stream once
+    let batches = SwanFlightSqlService::collect_record_batches(request).await?;
+    
+    if !batches.is_empty() {
```

Comment:
maybe we can write `if batches.is_empty()` then do the normal execution and return

otherwise we do appender, that way we don't indent this big code blocks